content/scripts/build.js

```diff
@@ -171,6 +171,25 @@ function broadcastWebsocketMessage(msg) {
   // console.log(`Sent to ${i} open clients`);
 }
 
+function getSummary(html) {
+  // TODO: Find a better way to do summaries!
+  // TODO: Ask Daniel Beck how his tool extracts summaries.
+  // TODO: There's a chicken and egg problem here. We'd like to use
+  //       the rendered HTML for extracting the summary, but the
+  //       summaries are used to create the rendered HTML!
+  // This is a simple implementation very loosely based on Kuma's approach,
+  // which is a mess. For now, let's just try our best. The summaries are
+  // mainly used (95% of the time) by the Kumascript macros for tool tips
+  // ("title" attribute) on links to their pages.
+  const $ = cheerio.load(html);
+  return (
+    $(".seoSummary").text() ||
+    $(".summary").text() ||
+    $("p").html() ||
+    ""
+  ).trim();
+}
```

Comment:
Yeah, this is pretty important actually. Even if it means we need to tangent on this a bit. The reason is that the sidebars make up a majority of the HTML payload.
Look at https://developer.mozilla.org/en-US/docs/Web/HTML/Element/embed for example. 
And look at https://htmltree.herokuapp.com/?url=https:%2F%2Fdeveloper.mozilla.org%2Fen-US%2Fdocs%2FWeb%2FHTML%2FElement%2Fembed
The sidebar's HTML amounts to 41KB and the document content only 21KB. 
Its because the sidebar is actually full of links and its full of those `title="{summar}"` attributes. 

And remember, Yari 2 is hydrated React so the HTML (which includes the sidebar HTML) is considered React "state" so it gets included in the SSR rendered HTML and then it's also made available as a JSON serialized blob which is shipped too. 

There are potential ideas for optimization and all. 
But basically, if the summaries are too large, it will potentially severely bloat the whole HTML document weight. 
Not much to do at this very point but worth keeping in mind.